pyroomacoustics/open_sofa_interpolate.py

```diff
@@ -0,0 +1,556 @@
+    directivity_pattern (int) : The specific pattern in the DIRPAT files are associated with id's , presented in the github document
+    source (Boolean) : If the DIRPAT files represnts one of source or receiver's
+    interpolate (Boolean) : Interpolate the FIR filter or not
+    no_of_points_fibo_sphere (int) : How many points should be there on fibonnacci sphere
+    azimuth_ (int) , colatitude_ : Orientation of the source/mic in the simulation environment.
+
+
+    """
+
+    def __init__(self,path,fs,directivity_pattern=2,source=False,interpolate=True,no_of_points_fibo_sphere=1200,azimuth_simulation=0,colatitude_simulation=0):
+
+        self.path=path
+        self.id = directivity_pattern
+        self.source = source
+
+
+
+        self.fs=fs
+
+        self.phi_sofa, self.theta_sofa, self.r_sofa, self.theta_16, self.sofa_msr_fir, self.no_of_nodes, self.samples_size_ir, self.no_of_lat = self.open_sofa_database(path=self.path,
+            fs=self.fs)
+
+        self.sofa_x,self.sofa_y,self.sofa_z=sph2cart(self.phi_sofa,self.theta_sofa,self.r_sofa)
+
+        self.interpolate = interpolate
+
+        #Rotation matrix
+
+        n_c = (colatitude_simulation)
+        n_a = (azimuth_simulation)
+        R_y = np.array([[np.cos(n_c), 0, np.sin(n_c)], [0, 1, 0], [-np.sin(n_c), 0, np.cos(n_c)]])
+        R_z = np.array([[np.cos(n_a), -np.sin(n_a), 0], [np.sin(n_a), np.cos(n_a), 0], [0, 0, 1]])
+        res = np.matmul(R_z, R_y)
+
+        if interpolate :
+            self.degree = 12
+
+            self.points = np.array(fibonacci_sphere(samples=no_of_points_fibo_sphere))
+            self.phi_fibo, self.theta_fibo, self.r_fibo = cart2sphere(self.points)
+
+
+            # All the computations are in radians for phi = range (0, 2*np.pi) and theta = range (0, np.pi)
+            # Just for numpy, the function accepts phi and theta in the opposite way
+
+            self.theta_sofa_np = self.phi_sofa
+            self.phi_sofa_np = self.theta_sofa
+            self.Ysh = cal_sph_basis(self.theta_sofa_np, self.phi_sofa_np, self.degree, self.no_of_nodes)
+
+            # Calculate spherical basis for the target grid (fibonacci grid)
+            self.theta_fibo_np = self.phi_fibo
+            self.phi_fibo_np = self.theta_fibo
+
+            self.Ysh_fibo = cal_sph_basis(self.theta_fibo_np, self.phi_fibo_np, self.degree, no_of_points_fibo_sphere)
+
+            # calculate pinv and voronoi cells for least square solution for the whole grid
+
+            self.Ysh_tilda_inv, self.w_ = calculation_pinv_voronoi_cells(self.Ysh, self.theta_16, self.no_of_lat)
+
+
+            self.freq_angles_fft = np.zeros((self.no_of_nodes, self.samples_size_ir), dtype=np.complex_) #N-point FFT
+            self.sh_coeffs_expanded_original_grid = 0
+            self.sh_coeffs_expanded_target_grid = 0
+
+            # Rotate the target grid (Fibonacci grid)
+
+            self.rotated_fibo_points = np.matmul(res, self.points.T)
```

Comment:
I see that you are currently rotating all the points of the interpolated grid.
Another solution is to rotate the incident points when you measure the directivity.
This would make it easier to share the interpolated grid in case multiple objects using the same directivity are used without having to recompute the interpolation for all of them.

Actually, I am now thinking that we could just store the interpolated grids with pyroomacoustics instead of loading from DIRPAT and interpolating every time... Thoughts ?